src/Albany_Application.cpp

```diff
@@ -2954,11 +2966,19 @@ Application::evaluateResidual_HessVecProd_pp(
     ParamVec params_l1 = param_array[l1];
     unsigned int num_cols_p_l1 = params_l1.size();
 
+    // Assumes that there is only one element block
+    int deriv_size = PHAL::getDerivativeDimensions<EvalT>(this, 0);
+    TEUCHOS_TEST_FOR_EXCEPTION(
+        num_cols_p_l1 > deriv_size,
+        std::runtime_error,
+            "\nError in Albany::Application::evaluateResidual_HessVecProd_pp  "
+            << "Number of parameters columns cannot be larger than the derivative size\n ");
+
     HessianVecFad p_val;
     for (unsigned int i = 0; i < num_cols_p_l1; i++) {
-      p_val = HessianVecFad(num_cols_p_l1, params_l1[i].baseValue);
+      p_val = HessianVecFad(deriv_size, params_l1[i].baseValue);
       p_val.fastAccessDx(i).val() = 1.0;
-      params_l1[i].family->setValue<PHAL::AlbanyTraits::HessianVec>(p_val);
+      params_l1[i].family->setValue<EvalT>(p_val);
```

Comment:
Why did you change this line?
If this is not needed, I would be in favor of keeping it as before or to change the other lines such as :

https://github.com/sandialabs/Albany/blob/d89979ca25d07aa7007b938f76742f3968c56883/src/responses/Albany_FieldManagerScalarResponseFunction.cpp#L532
and
https://github.com/sandialabs/Albany/blob/d89979ca25d07aa7007b938f76742f3968c56883/src/responses/Albany_FieldManagerScalarResponseFunction.cpp#L623
.

If those lines are consistent, it is easier to modify them all together if needed in the future.

Reply:
I can change it back, it's just that you have
`using EvalT =  PHAL::AlbanyTraits::HessianVec`
 at the beginning of the function and I thought it would make sense to use it.

Reply:
I'm in favor of using `EvalT` (shorter lines when you multiple terminal windows). But I agree that it's best if it's consistent. So I vote for using the typedef everywhere that it's possible. :)

Reply:
I +1 Luca's comment.

Reply:
OK, I did that. Merging...